### DISCUSSION TITLE
Why newline `\\` for latex in markdown needs `\\\\` when using `$$` in Github?

### DISCUSSION THREAD

**Discussion by sci-42ver** [community/community#121416] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

MathJax supports `\\` https://stackoverflow.com/a/32507429/21294350. 

However, `$$\begin{align*}A&\to B\\ C&\to D\end{align*}$$`

$$\begin{align*}A&\to B\\ C&\to D\end{align*}$$

doesn't work. `\\` works with [<span>```math...</span>](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions#writing-expressions-as-blocks). But normally the latter isn't supported in other applications.

---

It is weird we need to use `\\\\` like `$$\begin{align*}A&\to B\\\\ C&\to D\end{align*}$$`

$$
\begin{align*}A&\to B\\\\ C&\to D\end{align*}
$$

to render correctly. 

Is there one way to solve with this problem for `$$` elegantly?

**Comment by MrMadaraUchiha** [comment]
When writing LaTeX formulas in a Markdown document for display on GitHub, you may find the need to use two backslashes (\\) to indicate a line break instead of a single backslash (\). This is because Markdown itself uses a backslash to escape special characters, such as the backslash character, which is used in LaTeX to indicate a line break.

When you use $...$ to delimit a LaTeX expression in a Markdown document, Markdown processes the backslash, but LaTeX still needs it to understand line breaks. So, to prevent Markdown from consuming the first backslash, you need to use two of them (\\) so that LaTeX understands it as a line break.

Therefore, to ensure that LaTeX correctly interprets line breaks in a GitHub Markdown environment, you need to use \\\\ for each line break within an expression delimited by $...$. This ensures that Markdown correctly interprets the backslashes and passes them to LaTeX for proper formatting.

**Reply by inikishev** [reply]
Why doesn't this work?
```
$$B \\\\ C$$
```
$$B \\\\ C$$

**Comment by iv4n-ga6l** [comment, marked as answer]
In LaTeX, the double backslash "\" is used to create a newline within a document. However, when writing LaTeX within Markdown, such as in GitHub Markdown, you often need to escape the backslash character because Markdown uses it for its own formatting purposes.

So, when you want to create a newline in LaTeX within Markdown on GitHub, you need to use four backslashes "\\". This is because Markdown requires you to escape the backslash with another backslash, and then LaTeX requires its own double backslash for a newline. Hence, the total becomes four backslashes: two for Markdown and two for LaTeX.

For example, in GitHub Markdown, you would write:
<img width="445" alt="image" src="https://github.com/community/community/assets/75835662/588e0362-a156-41eb-85c9-5971146a424d">


This would render in LaTeX as:
<img width="443" alt="image" src="https://github.com/community/community/assets/75835662/5ce6a893-fd4c-4965-b349-b80668ff416a">

**Reply by sci-42ver** [reply]
It turns out that we need one `\n` after `\\` to make it work. Thanks for your example.

So the render well in github, the minimal example:
```
It is ...

$$\begin{align*}A&\to B\\
C&\to D\end{align*}$$

to ... 
```
It is ...

$$\begin{align*}A&\to B\\
C&\to D\end{align*}$$

to ... 

---

After normal formatting of latex,
```
It is ...

$$
\begin{align*}
A&\to B\\
C&\to D
\end{align*}
$$

to ... 
```
It is ...

$$
\begin{align*}
A&\to B\\
C&\to D
\end{align*}
$$

to ... 

---

notice we need _one newline between `$$` and other contents_ although this is not needed in vscode and stackedit. The original problem may be this lack of newline, so I _follow the doc with oneliner_ instead of the original multiline version. More specifically,  the following doesn't work.
```
It is ...
$$
\begin{align*}
A&\to B\\
C&\to D
\end{align*}
$$

to ... 
```
It is ...
$$
\begin{align*}
A&\to B\\
C&\to D
\end{align*}
$$

to ...

**Comment by sci-42ver** [comment]
But I have to say this 2 `\\` to function as `\` in Github markdown is very inconvenient since it is incompatible with VSCode and StackEdit.

For example, we need `$\\{1\\}$` $\\{1\\}$ to function as `$\{1\}$` but the former doesn't even render anything in VSCode.

---

Also GitHub markdown seems to not recognize indentation well.
```
- point 1

$$
\begin{align*}
A&\to B\\
C&\to D
\end{align*}
$$

  - point 1.1
```
- point 1

$$
\begin{align*}
A&\to B\\
C&\to D
\end{align*}
$$

  - point 1.1
```
- point 1

  $$
  \begin{align*}
  A&\to B\\
  C&\to D
  \end{align*}
  $$

  - point 1.1
```
- point 1

  $$
  \begin{align*}
  A&\to B\\
  C&\to D
  \end{align*}
  $$

  - point 1.1
```
- point 1

  - point 1.1
```
- point 1

  - point 1.1

---

Maybe it is not one good idea to use latex in Github markdown due to incompatibility.

**Reply by MrMadaraUchiha** [reply]
Here's an example of how you can format LaTeX equations in GitHub markdown in a way that's more compatible with different environments, including VSCode and StackEdit:

To insert an inline equation, you can use $`...`$. For example, $`f(x) = x^2`$.

For block equations, you can use ```math``` to delimit the block. For example:

```math
\begin{align*}
    A & \to B \\
    C & \to D
\end{align*}

**Reply by sci-42ver** [reply]
Do you mean `$...$`? Could you give the equation writing before rendered?

---

`math` is not supported in other platforms. So I will leave these problems there in Github and view in VSCode and StackEdit always.
